theano/sandbox/cuda/tests/test_driver.py

```diff
@@ -69,5 +69,24 @@ def test_nvidia_driver3():
     assert any([isinstance(node.op, cuda.GpuElemwise) for node in topo])
     assert theano.sandbox.cuda.use.device_number is not None
 
+
+def test_nvcc_cast():
+    """Test that the casting behaviour is correct.
+
+    Some versions of nvcc, in particular the one in 6.5.14, return an incorrect
+    value in this case.
+
+    Reported by Zijung Zhang at
+    https://groups.google.com/d/topic/theano-dev/LzHtP2OWeRE/discussion
+    """
+    var = theano.tensor.fvector()
+    f = theano.function([var], -1. * (var > 0), mode=mode_with_gpu)
+    if not numpy.allclose(f([-1, 0, 1]), [0, 0, -1]):
+        raise Exception(
```

Comment:
Do we want to always raise? I think it would prevent to run too much frequently as cuda 6.5 is installed at many places and it work correctly most of the time. What about doing this change in GpuElemwise, when the intenal is a Composite, where some of the intermediate variable are not float?